### PR TITLE
Add support for two new environment variables for `chandra_models` defaults

### PR DESCRIPTION
**Pull request by taldcroft**
## Description

This adds two new environment variables that impact the behavior:

- ``CHANDRA_MODELS_REPO_DIR`` as an alias for ``THERMAL_MODELS_DIR_FOR_MATLAB_TOOLS_SW``:
      override the default root for the chandra_models repository. The name defined by FOT Matlab tools is impossible
      to remember. :smile:
- ``CHANDRA_MODELS_DEFAULT_VERSION``: override the default repo version. 

Adding, ``CHANDRA_MODELS_DEFAULT_VERSION`` environment variable might be controversial but this greatly facilitates two important use cases:
- Setting version to a fixed version in unit tests (e.g. with ``monkeypatch``)
- Set to a developement branch to test a model file update with applications like yoshi where
   specifying a version would require a long chain of API updates.

## Interface impacts

None

### Unit tests

- [x] Mac

Independent check of unit tests by Jean
- [x] Linux

### Functional tests

@jeanconn extracted the FOT Matlab tools archive of `chandra_models`. This looks like below, where each directory is a shallow git repo with just the latest commit:
```
ska3-dev) ➜  chandra_aca git:(model-globs) ls /Users/aldcroft/Downloads/Thermal_Models
Archive/               chandra_models-3.43/   chandra_models-3.44/   chandra_models-3.46/   chandra_models-3.48/
chandra_models-3.41.1/ chandra_models-3.43.1/ chandra_models-3.45/   chandra_models-3.47/
```

Then I did the following in a dev ska with this repo installed:

```
In [11]: os.environ[
    ...:     "CHANDRA_MODELS_REPO_DIR"
    ...: ] = "/Users/aldcroft/Downloads/Thermal_Models/chandra_models-3.47"

In [12]: data, info = chandra_models.get_data("chandra_models/xija/aca/aca_spec.json")

In [13]: import json

In [14]: spec = json.loads(data)

In [15]: spec["name"]
Out[15]: 'aacccdpt'

In [16]: info
Out[16]: 
{'call_args': {'file_path': 'chandra_models/xija/aca/aca_spec.json',
  'version': None,
  'repo_path': 'None',
  'require_latest_version': False,
  'timeout': 5,
  'read_func': 'None',
  'read_func_kwargs': None},
 'version': '3.47',
 'commit': '5307bf8ad3a8588de3417b44f8eddf2a5a26a9e0',
 'data_file_path': '/Users/aldcroft/Downloads/Thermal_Models/chandra_models-3.47/chandra_models/xija/aca/aca_spec.json',
 'repo_path': '/Users/aldcroft/Downloads/Thermal_Models/chandra_models-3.47',
 'CHANDRA_MODELS_DEFAULT_VERSION': None,
 'CHANDRA_MODELS_REPO_DIR': '/Users/aldcroft/Downloads/Thermal_Models/chandra_models-3.47',
 'md5': '71cda7e5c2c5f72e5007f5d79014bd14'}

In [17]: del os.environ["CHANDRA_MODELS_REPO_DIR"]

In [18]: data, info = chandra_models.get_data("chandra_models/xija/aca/aca_spec.json")

In [19]: info
Out[19]: 
{'call_args': {'file_path': 'chandra_models/xija/aca/aca_spec.json',
  'version': None,
  'repo_path': 'None',
  'require_latest_version': False,
  'timeout': 5,
  'read_func': 'None',
  'read_func_kwargs': None},
 'version': '3.48',
 'commit': '68a58099a9b51bef52ef14fbd0f1971f950e6ba3',
 'data_file_path': '/Users/aldcroft/ska/data/chandra_models/chandra_models/xija/aca/aca_spec.json',
 'repo_path': '/Users/aldcroft/ska/data/chandra_models',
 'CHANDRA_MODELS_DEFAULT_VERSION': None,
 'CHANDRA_MODELS_REPO_DIR': None,
 'md5': '71cda7e5c2c5f72e5007f5d79014bd14'}
```

### CODE REVIEW

File: ska_helpers/paths.py [hunk 1]
```diff
@@ -15,20 +18,25 @@ def chandra_models_repo_path() -> Path:
...
+
+    - ``CHANDRA_MODELS_REPO_DIR``
+    - ``THERMAL_MODELS_DIR_FOR_MATLAB_TOOLS_SW``
 
     :returns: Path object
         Path to ``chandra_models`` repository
     """
+    for env_var in CHANDRA_MODELS_ROOT_ENV_VARS:
+        if env_var in os.environ:
+            return Path(os.environ[env_var])
```
Comment (jeanconn) [inline ska_helpers/paths.py:32]: Feels a little odd to handle this via an "early" `return` but it is a short function.

**Comment by jeanconn** [comment]
I note that I think https://github.com/sot/ska_sun/pull/31 needs to go at the same time.

File: ska_helpers/paths.py (hunk 1 above)
Comment (taldcroft) [reply ska_helpers/paths.py:32]: From what I've read early returns are generally a good thing and encouraged to reduce code complexity. Here is an opinion from the internet that must therefore be true. http://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html

Comment (taldcroft) [reply ska_helpers/paths.py:32]: It's worth saying that this is something new-ish to me and might not be reflected in earlier code / patterns.

**Pull request merged**